packages/tinacms-authjs/src/index.ts

```diff
@@ -133,10 +133,46 @@ const TinaCredentialsProvider = ({
       authenticate(databaseClient, credentials.username, credentials.password),
   })
 
+const NextAuthAuthentication = ({
+  nextAuthOptions,
+}: {
+  nextAuthOptions: AuthOptions
+}) => {
+  return {
+    isAuthenticated: async (req: any, res: any) => {
+      const session = await getServerSession(req, res, nextAuthOptions)
+
+      // @ts-ignore
+      if (!req.session) {
+        Object.defineProperty(req, 'session', {
+          value: session,
+          writable: false,
+        })
+      }
+
+      if (!session?.user?.name) {
```

Comment:
I think this should be isAuthorized and should actually check the role. Refer to https://github.com/tinacms/tinacms/blob/feat/self-hosted-improvements/packages/tinacms-authjs/src/index.ts#L98

Reply:
Updated